src/lib/tecs/TECS.cpp

```diff
@@ -778,13 +783,27 @@ void TECS::_setFastDescend(const float alt_setpoint, const float alt)
 {
 	if (_control_flag.airspeed_enabled && (_fast_descend_alt_err > FLT_EPSILON)
 	    && ((alt_setpoint + _fast_descend_alt_err) < alt)) {
-		_fast_descend = 1.f;
+		auto now = hrt_absolute_time();
+
+		if (_enabled_fast_descend_timestamp == 0U) {
+			_enabled_fast_descend_timestamp = now;
+		}
+
+		if ((now - _enabled_fast_descend_timestamp) < FAST_DESCEND_RAMP_UP_TIME) {
```

Comment:
I don't think that if() is needed, you already catch the case where now is more than FAST_DESCEND_RAMP_UP_TIME ahead of _enabled_fast_descend_timestamp with the constraint to max 1.